specs/agents/tracing-instrumentation-messaging.md

```diff
@@ -0,0 +1,103 @@
+## Messaging Systems
+
+The instrumentation of messaging systems includes:
+- Transaction creation on message reception, either as a child of the message-sending span, if the sending action is traced (meaning 
+distributed tracing support), or as root
+- Span creation for message sending action
+- Span creation on message *polling that occurs within a traced transaction*
+
+### Passive vs. active message reception
+
+Message send/publish events can only be captured as spans if occurring within a traced transaction.
+Message consumption can be divided into two types: passive, where you would implement a listener that is called once a message is available, 
+and active, where the queue/topic is being polled (blocking or non-blocking). 
+
+Passive consumption typically results in a `messaging` transaction and is pretty straightforward to trace - start at entry and end at exit. 
+
+Message polling can be done within a traced transaction, in which case it should result in a messaging span, or it can be the initiating 
+event for a message handling flow. Capturing polling spans is also mostly straightforward. 
+For polling-based transactions, our goal is to capture the message handling flow, which typically *starts after the polling action exits*, 
+returning a message. This may be tricky if the handling flow is not implemented within a well defined API. Use other agent implementation
+as reference.
+The agent should not create a transaction based on polling APIs if the polling action did not result with a message (as opposed to 
+polling spans, where we want to capture such as well).
+
+### Typing
+
+- Transactions: 
+  - `transaction.type`: `messaging`
+- Spans: 
+  - `span.type`: `messaging` 
+  - `span.subtype`: the name of the framework - e.g. `jms`, `kafka`, `rabbitmq` 
+  - `span.action`: `send`, `receive` or `poll`
+  
+### Naming
+
+Transaction and span names may* follow this pattern: `<MSG-FRAMEWORK> SEND/RECEIVE/POLL to/from <QUEUE-NAME>`.
```

Comment:
```suggestion
Transaction and span names SHOULD* follow this pattern: `<MSG-FRAMEWORK> SEND/RECEIVE/POLL to/from <QUEUE-NAME>`.
```

Reply:
[minor] I wonder if we may want to remove the framework name as we have it in the metadata. For Redis, we also don't prefix the span name with `Redis`, for example.

Reply:
I don't mind changing the name to something else. What do you think would work better but still indicate this is messaging-related **already in the waterfall view** (meaning - without opening span details)? 

HTTP transactions has the `HTTP` indication on them and DB spans has a dedicated icon.

I think it is important that the user can easily see it is related to messaging, especially when there is an expected corresponding child transaction that can be a bit delayed because of messaging async nature.

Reply:
Ideally, the UI would show a dedicated icon for messaging spans.

Reply:
I completely agree, that would be best! As for the spec, I will merge it as is and let's change once we have a better UI indication.